### DISCUSSION TITLE
扫描不出可用ip

### DISCUSSION THREAD

**Discussion by WB967** [GoodCoder666/GoogleTranslate_IPFinder#19] [General]
大佬你好，我是win11系统，下载了checker-win-x64，扫描后无ip，视频没有剪辑，40''~3'06''全是等待时间，请直接拖动。
本地有clash，但是未开启系统代理，应该不是clash影响的。
https://www.aliyundrive.com/t/N8GgMSwPaOCpKmZdvMxO

**Comment by GoodCoder666** [comment]
从技术角度解释：
- 现在使用默认参数只会扫描 `142.250.0.0/15` 段的 IP，若开启优化则总共会扫描 1024 个 IP，32 线程理论需要 80 秒左右的时间完成扫描，实际用时一般会大于这个值。
- 由于种种原因，现在可用的 IP 少之又少，以至于 `142.250.0.0/15` 段中无可用 IP，所以扫描不出结果。
- 建议直接使用默认的 IP 库，选择“测速“得到可用 IP，目前普遍可用的应该是 `108.177.97.100`。

**Comment by GoodCoder666** [comment]
另外如果真的要使用扫描，建议使用如下配置：
- 最大线程数：200
- 响应时间上限：1.5s
- 启用 扩展扫描

我后续会尝试针对现在的网络环境重做一下扫描功能，毕竟是远古功能了。

**Comment by WB967** [comment]
谢谢大佬答疑

**Comment by myunco** [comment]
不知道什么原因，感觉几乎所有IP都不能用了。我一个稳定用了半年多的ipv6地址，今天发现不能用了，扫描一下发现全炸

**Comment by GoodCoder666** [comment]
目前已经不是程序自身的问题了，所以我移到了这个 Discussion。

**Comment by GoodCoder666** [comment]
> 不知道什么原因，感觉几乎所有IP都不能用了。我一个稳定用了半年多的ipv6地址，今天发现不能用了，扫描一下发现全炸

@myunco 可以尝试导入在线的 IPv6 库重新测速，找到能用的

或者用 IPv4，提供几个我这边能用的：

```
108.177.97.100
216.239.32.40
74.125.196.113
64.233.189.191
```

**Reply by GoodCoder666** [reply]
补充一个，`142.251.173.90`

不过这个貌似不太稳定，好像昨天不能用今天突然就能用了

**Comment by GoodCoder666** [comment]
优化了一下扫描逻辑，可以尝试用最新的 1.9.0 版本扫描。

**Comment by GoodCoder666** [comment]
加了个**不骗人的**进度条（最新版 1.10.0），也许可以解决扫描等待不耐烦的问题？

**Reply by myunco** [reply]
建议再加个 "关于" 说明一下作者/项目地址什么的，方便使用者知道软件来源（可能有人从其他地方下载想找原作者找不到）。

**Comment by GoodCoder666** [comment]
> 建议再加个 "关于" 说明一下作者/项目地址什么的，方便使用者知道软件来源（可能有人从其他地方下载想找原作者找不到）。

这样会影响整体界面简洁性，况且我貌似没发现有其他地方可以下载到？

**Comment by GoodCoder666** [comment]
建议大家更新到最新的 1.10.1，测试速度大幅提升。

另外我发现 IPv6 非常不稳定，没有任何一个 IPv6 能保证连续五次访问都成功。我在新版的程序中预留了一个参数作为稳定性阈值（至少连续多少次访问成功才行），但是现在没有启用（或者说，设置为 1），想征求一下大家的意见。IPv4 目前没有发现此问题。
